packages/flutter/lib/src/gestures/long_press.dart

```diff
@@ -58,3 +151,105 @@ class LongPressGestureRecognizer extends PrimaryPointerGestureRecognizer {
   @override
   String get debugDescription => 'long press';
 }
+
+/// Recognizes long presses that can be subsequently dragged around.
+///
+/// Similar to a [LongPressGestureRecognizer] where a press has to be held down
+/// at the same location for a long period of time. However, after
+/// [onLongPressDown] is triggered after the hold threshold, drags will not
+/// subsequently cancel the gesture while it's still held. The [onLongPressDrag]
+/// callback will be called as the drag moves.
+///
+/// See also:
+///
+///  * [LongPressGestureRecognizer], which cannot be dragged during the gesture.
+class LongPressDragGestureRecognizer extends PrimaryPointerGestureRecognizer {
+  /// Creates a long-press-drag gesture recognizer.
+  ///
+  /// Consider assigning the [onLongPressDrag] callback after creating this object.
+  LongPressDragGestureRecognizer({ Object debugOwner }) : super(
+    deadline: kLongPressTimeout,
+    // Since it's a drag gesture, no travel distance will cause it to get
+    // rejected after the long-press is accepted.
+    postAcceptSlopTolerance: null,
+    debugOwner: debugOwner,
+  );
+
+  bool _longPressAccepted = false;
+
+  Offset _longPressOrigin;
+
+  Duration _longPressDownTimestamp;
+
+  /// Called when a long press gesture has been recognized.
+  GestureLongPressDragDownCallback onLongPressDown;
```

Comment:
Ummmm actually on second thought, I think I'm going to rename all this stuff to onLongPressDragStart etc. "Down" is a bit misleading since it's triggered after a time delay, rather than when the pointer is down (which is in the past).